Design/Instrument-2.0/ScanningInstruments_Design.md

```diff
@@ -0,0 +1,170 @@
+
+### Implementation
+
+For more details on the rollout plan see the document [Scanning Instruments Plan](ScanningInstruments_Plan.md).
+
+For this design to be implemented the use of `SpectrumInfo` and `DetectorInfo` will need to be rolled out in Mantid. The rollout for [`SpectrumInfo`](https://github.com/mantidproject/mantid/issues/17743) has been started. This will be a significant amount of work to complete with around 100 algorithms to update.
+
+### New DetectorInfo Signatures
+
+Currently detector info public methods look like the following.
+
+```cpp
+DetectorInfo(boost::shared_ptr<const Geometry::Instrument> instrument,
+           Geometry::ParameterMap *pmap = nullptr);
+
+bool isMonitor(const size_t index) const;
+bool isMasked(const size_t index) const;
+double l2(const size_t index) const;
+double twoTheta(const size_t index) const;
+double signedTwoTheta(const size_t index) const;
+Kernel::V3D position(const size_t index) const;
+Kernel::Quat rotation(const size_t index) const;
+
+void setPosition(const size_t index, const Kernel::V3D &position);
+void setRotation(const size_t index, const Kernel::Quat &rotation);
+
+void setPosition(const Geometry::IComponent &comp, const Kernel::V3D &pos);
+void setRotation(const Geometry::IComponent &comp, const Kernel::Quat &rot);
+```
+
+The following methods will be given require a time index access, in addition to the detector index.
+
+```cpp
+DetectorInfo(boost::shared_ptr<const Geometry::Instrument> instrument,
+           Geometry::ParameterMap *pmap = nullptr);
+
+bool isMonitor(const size_t index) const;
+bool isMasked(const size_t index) const;
+double l2(const size_t index) const;
+double l2(const size_t index, const size_t timeIndex) const;
+double twoTheta(const size_t index) const;
+double twoTheta(const size_t index, const size_t timeIndex) const;
+double signedTwoTheta(const size_t index) const;
+double signedTwoTheta(const size_t index, const size_t timeIndex) const;
+Kernel::V3D position(const size_t index) const;
+Kernel::V3D position(const size_t index, const size_t timeIndex) const;
+Kernel::Quat rotation(const size_t index) const;
+Kernel::Quat rotation(const size_t index, const size_t timeIndex) const;
+
+void setPosition(const size_t index, const Kernel::V3D &position);
+void setPosition(const size_t index, const Kernel::V3D &position, const size_t timeIndex);
+void setRotation(const size_t index, const Kernel::Quat &rotation);
+void setRotation(const size_t index, const Kernel::Quat &rotation, const size_t timeIndex);
+
+void setPosition(const Geometry::IComponent &comp, const Kernel::V3D &pos);
+void setPosition(const Geometry::IComponent &comp, const Kernel::V3D &pos, const size_t timeIndex);
+void setRotation(const Geometry::IComponent &comp, const Kernel::Quat &rot);
+void setRotation(const Geometry::IComponent &comp, const Kernel::Quat &rot, const size_t timeIndex);
```

Comment:
This forwards to the instrument tree, component positions would not be stored in `DetectorInfo`, so I do not know how this could be implemented?